electron/src/lib/CoreProtocol.ts

```diff
@@ -40,52 +40,59 @@ enum ProtocolCommand {
 }
 
 export class CustomProtocolHandler {
-  public hashLocation: string = '';
+  hashLocation: string = '';
+  private readonly windowManager = WindowManager;
 
-  async dispatcher(url?: string) {
+  async dispatchDeepLink(url?: string): Promise<void> {
     if (typeof url === 'undefined' || !url.startsWith(CORE_PROTOCOL_PREFIX) || url.length > CORE_PROTOCOL_MAX_LENGTH) {
       return;
     }
 
     const route = new URL(url);
 
-    switch (route.host) {
-      case ProtocolCommand.START_SSO_FLOW: {
-        if (typeof route.pathname === 'string') {
-          logger.log('Starting SSO flow...');
-          const code = route.pathname.trim().substr(1);
-          await app.whenReady();
-          WindowManager.sendActionAndFocusWindow(EVENT_TYPE.ACCOUNT.SSO_LOGIN, code);
-        }
-        break;
-      }
-      default: {
-        const location = route.href.substr(CORE_PROTOCOL_PREFIX.length);
-        this.hashLocation = `/${location}`;
-        WindowManager.sendActionToPrimaryWindow(EVENT_TYPE.WEBAPP.CHANGE_LOCATION_HASH, this.hashLocation);
-        break;
+    if (route.host === ProtocolCommand.START_SSO_FLOW) {
+      await this.handleSSOLogin(route);
+    } else {
+      this.forwardHashLocation(route);
+    }
+  }
+
+  private forwardHashLocation(route: URL): void {
+    const location = route.href.substr(CORE_PROTOCOL_PREFIX.length);
+    this.hashLocation = `/${location}`;
+    this.windowManager.sendActionToPrimaryWindow(EVENT_TYPE.WEBAPP.CHANGE_LOCATION_HASH, this.hashLocation);
+  }
+
+  private async handleSSOLogin(route: URL): void {
+    if (typeof route.pathname === 'string') {
+      logger.log('Starting SSO flow...');
+      const code = route.pathname.trim().substr(1);
+      try {
+        await this.windowManager.sendActionAndFocusWindow(EVENT_TYPE.ACCOUNT.SSO_LOGIN, code);
+      } catch (error) {
+        logger.error(`Cannot start SSO flow: ${error.message}`, error);
       }
     }
   }
 
-  public registerCoreProtocol() {
+  public async registerCoreProtocol(): void {
```

Comment:
```suggestion
  public async registerCoreProtocol(): Promise<void> {
```

Reply:
Why do you want to declare this one as `async`?

Reply:
We should remove `async` for `registerCoreProtocol` because it's not `async`. Only the callbacks from the listeners that it creates are async.